src/paillier.rs

```diff
@@ -272,4 +306,133 @@ mod test {
             paillier_keygen_produces_good_primes()
         }
     }
+
+    fn random_message(
+        rng: &mut (impl CryptoRng + RngCore),
+        encryption_key: &PaillierEncryptionKey,
+    ) -> BigNumber {
+        let range: BigNumber = encryption_key.n() / 2;
```

Comment:
Should this be `(n - 1) / 2`? Depending on how `BigNumber` rounds it might affect things.

Reply:
Good catch, yes!